### ISSUE TITLE
3.14.0a2 rebuild needed for 0x03070000 limited api c extension

### ISSUE THREAD

**Issue by MrBitBucket** [python/cpython#127191]
# Bug report

### Bug description:

An existing C api extension compiled with Python 3.7-3.11 behaves differently when run in Python 3.14.0a2. Causing a reference count change where none is expected. The python test code looks like this

```python
from sys import getrefcount
from _rl_accel import unicode2T1

utext = 'This is the end of the world'

class Font:
    def __init__(self,name):
        self.fontName = name
        self.substitutionFonts = []
        self.encName = 'utf8'

font = Font('Helvetica')

defns = 'utext font font.encName font.fontName font.substitutionFonts'.split()
rc0 = [getrefcount(eval(x,globals())) for x in defns]
print(rc0)
unicode2T1(utext,[font]+font.substitutionFonts)
rc1 = [getrefcount(eval(x,globals())) for x in defns]
print(rc1)
if rc1!=rc0:
    print('!!!!! reference counts changed !!!!!')
```

The C function unicode2T1 is not expected to change any of the refcounts checked, but if built with python 3.7-3.11
we see a decrease in the refcount for font.encName when running in python 3.14.0a2. 

If built with python 3.14.0a2 the reference count decrease is not seen in the test. It seems that my code is broken in terms of the abi3 guarantee.

To ease checking I built a bash script to checkout and build the test environments; this can be found [here](https://github.com/MrBitBucket/rl-accel-mirror/blob/master/tests/runner.sh).

I run this like 
```sh
EARLYPYTHON=~/bin/python311 LATEPYTHON=~/bin/python314 tests/runner.sh
```

Sorry this example is so large, but at least reportlab is not required. We do have a python version of this code [rl_accel.py](https://hg.reportlab.com/hg-public/reportlab/file/c299e65db01b/src/reportlab/lib/rl_accel.py)

### CPython versions tested on:

3.9, 3.10, 3.11, 3.12, 3.13, 3.14

### Operating systems tested on:

Linux

### Related

[discuss.python.org/t/3-14-0a2-rebuild-needed-for-0x03070000-limited-api-c-extension/72006](https://discuss.python.org/t/3-14-0a2-rebuild-needed-for-0x03070000-limited-api-c-extension/72006)

**Comment by MrBitBucket** [comment]
I think this is expected now that the immortality of some/all globals is happening in Python >= 3.12. Probably I just need to build new wheels using a late python.

**Comment by picnixz** [comment]
I'm not sure we can do anything on our side since it affects 3rd party extensions and a security-branch only. However I don't know the exact policy concerning that kind of breaking change hence I'm asking @iritkatriel or @vstinner.

**Comment by ZeroIntensity** [comment]
As far as I know, the actual reference count of an object is an implementation detail, not part of the stable ABI.

**Comment by vstinner** [comment]
Can you try to write a reproducer which doesn't depend on third party code?

**Comment by vstinner** [comment]
Ah, this issue is also discussed at: https://discuss.python.org/t/3-14-0a2-rebuild-needed-for-0x03070000-limited-api-c-extension/72006

**Comment by erlend-aasland** [comment]
Based on the discussion over at DPO, it sounds to me we can close this as "not planned".

**Comment by ZeroIntensity** [comment]
Yeah, reference counts aren't part of the stable ABI. I wonder if we should add a `sys._is_immortal` for situations like this, similar to how we have `sys._is_interned`.
